standards/mobile_app_standards.md

```diff
@@ -0,0 +1,135 @@
+
+There are many such cross platform technology stacks, but some are obscure or are nearing obsolescence. This is still a too broad and fast moving area for us to as yet mandate solid standards, so you are therefore strongly advised to read the accompanying Mobile Application Guidance before considering a technology stack for developing cross platform apps.
+
+In the past one common technique used to build cross platform apps was to actually run the app inside a hidden browser on the device, with such apps often being called "hybrid" apps. However, that is not the approach adopted by any of latest and more popular frameworks, and it is increasingly challenged by ever stricter adoption of "single origin" security restrictions in more modern browser engines. A mobile app rendering local content automatically uses up one "origin", to be precise, before it even starts to interact with any \"remote\" systems, plus single origin restrictions disallow delivering up local content via AJAX calls. Therefore, such frameworks including Apache Cordova, the closely related PhoneGap (which is Adobe's own Cordova variant) and Cordova-based Ionic (though Ionic also now has a React based version) must now be regarded as legacy and **not used for new projects**. It is still just about possible to work around such restrictions using the above frameworks, but the approaches then become painful and convoluted, for example Cordova-based Ionic does so by using a local web server built into the app. **Avoid these "hybrid" approaches for building a cross-platform app within a browser.**
+
+Note that cross platform app development, where in any way feasible, **is always preferable to native app development, even when we are currently targeting only a single platform (say Apple) and range of associated devices**. That is because such apps ought to be easily adaptable to run on a different platform, for example if our selected choice of devices changes, or if we decide to make the app, or a version of the app, available on more devices. Furthermore, investment, in terms of time, money and "hard lessons learned", in cross platform app development is more likely to be applicable across the whole range of possible future apps than if we focus on a single native app technology.
+
+### Option 3: Build a Native App
+
+As a final fallback option it is therefore permissible to develop a native application. **Should the app be intended for anything other than purely and permanently internal use, this will involve developing and maintaining two entirely separate versions of the app, using the technologies native to the given platform: meaning using the Swift language and the Xcode IDE for Apple, and either the Java or Kotlin languages for Android** (Android Studio is the free and recommended IDE for such development but, as this is a much more open eco-system, it cannot be mandated).
```

Comment:
I honestly can't think of a justification for ever doing this.

In the extremely unlikely event that it is ever necessary, I'd prefer to handle it as an exception from the standard rather than having it as an option within thestandard.

Reply:
Yes, it would be pretty hideous. I'll change to briefly state something on the above lines. Have we got an exception process at the moment?